src/connection.ts

```diff
@@ -1586,16 +1610,7 @@ export class Connection extends EventEmitter {
   private bumpIdle (): void { this.lastActive = new Date() }
 
   private addTotalReceived (value: Long): void {
-    const result = checkedAdd(this._totalReceived, value)
-    if (result.overflow) {
```

Comment:
why is this check being removed? It was here because we're representing the numbers as Longs and STREAM technically could overflow that if enough is received on a connection

Reply:
That's the problem. After `shouldFulfill` resolves and the packet is accounted for as fulfilled, if totalReceived overflows _and throws_, then the packet will be rejected.

And, even if I check that `totalReceived` doesn't overflow before calling `shouldFulfill`, other packets could be received in the interim that then cause it to overflow.

What's the best solution here? Should I add "holds" on the total received to prevent this?

Reply:
I changed it back so it rejects the packet on the overflow.

I'm still not sure how I feel about this, because while it seems bad to credit the money in an external system, fulfill the packet, but incorrectly reflect that in `totalReceived`, it might be worse to credit the money in the external system, reject the packet, and correctly reflect `totalReceived`. (Since for this use case, the external system is probably the source of truth, rather than `totalReceived`).

That said -- it's definitely an edge case and a deployment should probably be configured to prevent fulfilling anywhere close to such a large amount.